### PR TITLE
shelter: add rpm and deb spec for package build

### PR DESCRIPTION
**Pull request by zhiminghufighting**
Fixes: #502
Signed-off-by: zhiminghufighting <13764765163@163.com>

### CODE REVIEW

File: shelter/dist/rpm/shelter.spec [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+popd
+
+%install
+install -d -p %{buildroot}%{BIN_DIR}
+install -p -m 755 %{name}/%{name} %{buildroot}%{BIN_DIR}
+
+%files
+%{BIN_DIR}/%{name}
+
+%changelog
```
Comment (hustliyilin) [inline shelter/dist/rpm/shelter.spec:53]: Merge the two changelogs into one.

File: shelter/dist/deb/debian/changelog [hunk 2]
```diff
@@ -0,0 +1,11 @@
+shelter (0.6.0-1) unstable; urgency=low
```
Comment (hustliyilin) [inline shelter/dist/deb/debian/changelog:1]: Merge the two changelogs into one.

File: shelter/dist/deb/debian/control [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Source: shelter
+Section: devel
+Priority: extra
+Maintainer: zhiming hu <zhiming.hu@intel.com>
+Build-Depends: debhelper (>=9)
```
Comment (hustliyilin) [inline shelter/dist/deb/debian/control:5]: You should find a clean ubuntu container and build shelter again and add the Build-Depends please.

File: shelter/dist/rpm/shelter.spec [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Name: shelter
+Version: 0.6.0
+Release: %{centos_base_release}%{?dist}
+Summary: shelter is designed as a remote attestation tool for customer to verify if their workloads are loaded in a specified intel authorized sgx enclaved.
+
+Group: Development/Tools
+License: Apache License 2.0
+URL: https://github.com/alibaba/%{PROJECT}
+Source0: https://github.com/alibaba/%{PROJECT}/archive/v%{version}.tar.gz
+
```
Comment (hustliyilin) [inline shelter/dist/rpm/shelter.spec:16]: You should find a clean ubuntu container and build shelter again and add the BuildRequirements please.

**Comment by hustliyilin** [comment]
Please add make shelter in the top Makefile of the inclavare containers project.

**Review by haosanzi — COMMENTED** [review]
Please change `inclavare-containers/Makefile` too.

File: shelter/dist/Makefile [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+RELEASE_TARBALL_EXIST := $(shell if [ -f $(RELEASE_TARBALL_FILE) ]; then echo "y"; else echo "n"; fi;)
+
+release-tarball:
+ifneq ($(RELEASE_TARBALL_EXIST), y)
+	@mkdir -p $(RPMBUILD_DIR)/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+	@wget -P $(RPMBUILD_DIR)/SOURCES $(RELEASE_TARBALL_URL)
+endif
+
+rpm: release-tarball
+	@rpmbuild -D "_topdir $(RPMBUILD_DIR)" -ba rpm/sgx-tools.spec
```
Comment (haosanzi) [inline shelter/dist/Makefile:14]: Still `rpm/sgx-tools.spec`?

Comment (zhiminghufighting) [reply shelter/dist/Makefile:14]: Sorry, i will correct it.

File: shelter/dist/rpm/shelter.spec (hunk 1 above)
Comment (zhiminghufighting) [reply shelter/dist/rpm/shelter.spec:58]: done

File: shelter/dist/deb/debian/changelog (hunk 2 above)
Comment (zhiminghufighting) [reply shelter/dist/deb/debian/changelog]: done

File: shelter/dist/deb/debian/control (hunk 3 above)
Comment (zhiminghufighting) [reply shelter/dist/deb/debian/control]: Add dependence lib into Build-Depends

File: shelter/dist/rpm/shelter.spec (hunk 4 above)
Comment (zhiminghufighting) [reply shelter/dist/rpm/shelter.spec:16]: Add dependence lib into BuildRequires

File: shelter/dist/rpm/shelter.spec (hunk 1 above)
Comment (zhiminghufighting) [reply shelter/dist/rpm/shelter.spec:58]: change package version from 0.6.0 to 0.6.1

**Comment by zhiminghufighting** [comment]
> Please add make shelter in the top Makefile of the inclavare containers project.

Add shelter into IC Makefile component list

File: shelter/dist/deb/debian/changelog [hunk 6]
```diff
@@ -0,0 +1,5 @@
+shelter (0.6.1-1) unstable; urgency=low
+
+  * Initial release.
+
+ -- Hu Zhiming <zhiming.hu@intel.com>  Sun, 07 2021 2020 13:48:35 +0000
```
Comment (hustliyilin) [inline shelter/dist/deb/debian/changelog:5]: `Sun, 07 2021 2020` is not a valid date, please fix it.

File: shelter/dist/rpm/shelter.spec [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+%install
+install -d -p %{buildroot}%{BIN_DIR}
+install -p -m 755 %{name}/%{name} %{buildroot}%{BIN_DIR}
+
+%files
+%{BIN_DIR}/%{name}
+
+%changelog
+* Sun 07 2021 Zhiming Hu <zhiming.hu@intel.com> - 0.6.1
```
Comment (hustliyilin) [inline shelter/dist/rpm/shelter.spec:59]: `Sun, 07 2021` is not a valid date, please fix it.

**Review by hustliyilin — APPROVED** [review]
Good Work!  But we also need to wait all CI/CD test cases successfully passing

File: Makefile [hunk 8]
```diff
@@ -1,7 +1,7 @@
 .PHONY: all install clean uninstall package
 
 export INCLAVARE_CONTAINERS_VERSION := $(shell cat ./VERSION)
-components := rune shim epm sgx-tools
+components := rune shim epm sgx-tools shelter
```
Comment (jiazhang0) [inline Makefile:4]: It is still unclear that the build failure happens from a scratch environment, such as:

```
CGO_ENABLED=1 go build "-mod=vendor" -buildmode=pie   -ldflags "" -o /root/dev/gopath/src/github.com/alibaba/inclavare-containers/shelter/racommand.o .
# github.com/inclavare-containers/shelter/remoteattestation
/usr/bin/ld: cannot find -lsgx_dcap_quoteverify
collect2: error: ld returned 1 exit status
Makefile:60: recipe for target '/root/dev/gopath/src/github.com/alibaba/inclavare-containers/shelter/racommand.o' failed
```

Comment (jiazhang0) [reply Makefile:4]: This issue is resolved now.

**Comment by jiazhang0** [comment]
Hold on this PR until enclave tls is merged.

**Pull request closed without merging**